lib/manager.js

```diff
@@ -401,6 +401,7 @@ Manager.prototype.destroy = function(socket){
 Manager.prototype.packet = function(packet){
   debug('writing packet %j', packet);
   var self = this;
+  if (packet.query && packet.type == 0) packet.nsp += '?' + packet.query;
```

Comment:
I think `query` can be in `data` rather than appending to `nsp`. Any thoughts about that ? cc @rauchg